testsuite/MDAnalysisTests/test_reader_api.py

```diff
@@ -175,6 +176,24 @@ def sl():
             return list(self.reader[1.2:2.5:0.1])
         assert_raises(TypeError, sl)
 
+    def test_getitem_list_ints(self):
+        sl = [0, 1, 4, 5]
+        self._check_slice(sl)
+
+    def test_getitem_array_ints(self):
+        sl = np.array([0, 1, 4, 5])
+        self._check_slice(sl)
+
+    def test_list_TE(self):
+        def sl():
+            return list(self.reader[[0, 'a', 5, 6]])
+        assert_raises(TypeError, sl)
+
+    def test_array_TE(self):
+        def sl():
+            return list(self.reader[np.array([1.2, 3.4, 5.6])])
+        assert_raises(TypeError, sl)
+
```

Comment:
Can you add a test for each of:

1. Repeated indices?
2. Out-of-order indices?